src/zkevm_specs/evm/execution/storage.py

```diff
@@ -0,0 +1,113 @@
+from ..instruction import Instruction, Transition
+from ..opcode import Opcode
+from ..table import CallContextFieldTag, TxContextFieldTag
+from .storage_gas import (
+    COLD_SLOAD_COST,
+    WARM_STORAGE_READ_COST,
+    SLOAD_GAS,
+    SSTORE_SET_GAS,
+    SSTORE_RESET_GAS,
+    SSTORE_CLEARS_SCHEDULE,
+)
+
+
+def sload(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+    instruction.constrain_equal(opcode, Opcode.SLOAD)
+
+    tx_id = instruction.call_context_lookup(CallContextFieldTag.TxId)
+    rw_counter_end_of_reversion = instruction.call_context_lookup(
+        CallContextFieldTag.RwCounterEndOfReversion
+    )
+    is_persistent = instruction.call_context_lookup(CallContextFieldTag.IsPersistent)
+    callee_address = instruction.call_context_lookup(CallContextFieldTag.CalleeAddress)
+
+    storage_key = instruction.stack_pop()
+
+    instruction.account_storage_read(callee_address, storage_key)
+
+    is_warm_new, is_warm = instruction.add_account_storage_to_access_list_with_reversion(
+        tx_id, callee_address, storage_key, is_persistent, rw_counter_end_of_reversion
+    )
+
+    instruction.stack_push()
+
+    # TODO: Use intrinsic gas (EIP 2028, 2930)
+    dynamic_gas_cost = WARM_STORAGE_READ_COST if is_warm == 1 else COLD_SLOAD_COST
+
+    instruction.step_state_transition_in_same_context(
+        opcode,
+        rw_counter=Transition.delta(8),
+        program_counter=Transition.delta(1),
+        stack_pointer=Transition.delta(0),
+        state_write_counter=Transition.delta(1),
+        dynamic_gas_cost=dynamic_gas_cost,
+    )
+
+
+def sstore(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+    instruction.constrain_equal(opcode, Opcode.SSTORE)
+
+    tx_id = instruction.call_context_lookup(CallContextFieldTag.TxId)
+    rw_counter_end_of_reversion = instruction.call_context_lookup(
+        CallContextFieldTag.RwCounterEndOfReversion
+    )
+    is_persistent = instruction.call_context_lookup(CallContextFieldTag.IsPersistent)
+    callee_address = instruction.call_context_lookup(CallContextFieldTag.CalleeAddress)
+
+    storage_key = instruction.stack_pop()
+    value_new = instruction.stack_pop()
+
+    _, value_prev, txid, original_value = instruction.account_storage_write_with_reversion(
+        callee_address, storage_key, is_persistent, rw_counter_end_of_reversion
+    )
+    instruction.constrain_equal(tx_id, txid)
+
+    is_warm_new, is_warm = instruction.add_account_storage_to_access_list_with_reversion(
+        tx_id, callee_address, storage_key, is_persistent, rw_counter_end_of_reversion
+    )
+
+    gas_refund, gas_refund_prev = instruction.tx_refund_write_with_reversion(
+        tx_id, is_persistent, rw_counter_end_of_reversion
+    )
+    gas_refund_new = gas_refund_prev
+    if value_prev != value_new:
```

Comment:
I am thinking we could use `is_equal` flag instead of branches, which better reflects circuit, and also better reflects  yellowpaper's long description. But it's okay to just keep as it now, since there are already a lot thing hard to reflect circuit's behavior.

Reply:
opened an issue: https://github.com/appliedzkp/zkevm-specs/issues/123

I will work on it as my following PR